src/Dialect/ONNX/Rewrite.cpp

```diff
@@ -102,6 +103,36 @@ bool areProducedByTransposeOp(ValueRange values) {
   });
 }
 
+// Create a DenseElementsAttr based on the shape of type.
+DenseElementsAttr createDenseElementsAttrFromShape(PatternRewriter &rewriter,
+    Value value, int64_t start = 0, llvm::Optional<int64_t> end = llvm::None) {
+
+  auto inType = value.getType().cast<ShapedType>();
+  assert(inType.hasRank() && "inType must be ranked");
+  auto shape = inType.getShape();
+  int64_t rank = inType.getRank();
+
+  int64_t endValue = end.has_value() ? end.value() : rank;
+
```

Comment:
would it make sense to have an assert here for sizes, or better, it could be added in the verifier (apologies if it is already there)